torch/nn/quantized/modules/conv.py

```diff
@@ -49,80 +48,48 @@ class Conv2d(_ConvNd):
-                                     bias=True,
-                                     padding_mode=padding_mode)
-        del self.weight
-        del self.bias
-
+                "Currently only zero-padding is supported by quantized conv")
+        if in_channels % groups != 0:
+            raise ValueError('in_channels must be divisible by groups')
+        if out_channels % groups != 0:
+            raise ValueError('out_channels must be divisible by groups')
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.kernel_size = kernel_size
+        self.stride = _pair(stride)
+        self.padding = _pair(padding)
+        self.dilation = _pair(dilation)
+        self.transposed = False
+        self.output_padding = 0
+        self.groups = groups
+        self.padding_mode = padding_mode
         qweight = torch._empty_affine_quantized(
             [out_channels, kernel_size[0], kernel_size[1],
              in_channels // self.groups],
             scale=1, zero_point=0, dtype=torch.qint8)
-        qbias = torch._empty_affine_quantized([out_channels],
-                                              scale=1, zero_point=0,
-                                              dtype=torch.qint32)
-        self.register_buffer('_packed_weight', torch.ops.quantized.fbgemm_conv_prepack(qweight.permute([0, 2, 3, 1]),
-                             self.stride, self.padding, self.dilation, self.groups))
-        self.register_buffer('bias', qbias)
-        self.register_buffer('scale', torch.tensor([1.0], dtype=torch.double))
-        self.register_buffer('zero_point', torch.tensor([0], dtype=torch.long))
-
-    @property
-    def weight(self):
-        return torch.ops.quantized.fbgemm_conv_unpack(self._packed_weight).permute([0, 3, 1, 2])
-
-    @weight.setter
-    def weight(self, w):
-        self._packed_weight = torch.ops.quantized.fbgemm_conv_prepack(w.permute([0, 2, 3, 1]),
-                                                                      self.stride,
-                                                                      self.padding,
-                                                                      self.dilation,
-                                                                      self.groups)
-
-    @property
-    def scale(self):
-        return self._scale.item()
-
-    @scale.setter
-    def scale(self, s):
-        if isinstance(s, torch.Tensor):
-            self._scale = s
-        else:
-            self._scale = torch.tensor([s], dtype=torch.double)
+        self.set_weight(qweight)
+        self.bias = torch._empty_affine_quantized([out_channels],
+                                                  scale=1, zero_point=0,
+                                                  dtype=torch.qint32)
+        self.scale = 1.0
+        self.zero_point = 0
 
-    @property
-    def zero_point(self):
-        return self._zero_point.item()
+    def set_weight(self, w):
```

Comment:
Is there an advantage of using explicit setters/getters instead of the properties? I feel like we are stepping off the common pattern, where weight/bias are just variables, not methods.

Reply:
are properties breaking with TorchScript? but I agree - would be nice to make them work (I think it should be possible to disambiguate properties and fields because properties are on the class level, not object level)

Reply:
Yes, properties are unsupported in TorchScript, I can file an issue